spec/index.bs

```diff
@@ -1191,131 +1209,176 @@ access to it before garbage collection by calling {{GPUBuffer/destroy()}}.
+content via {{ArrayBuffer}}s that represent part of the {{GPUBuffer}}'s
+allocations. Mapping a {{GPUBuffer}} is requested asynchronously with
+{{GPUBuffer/mapAsync}} so that the user agent can ensure the GPU
+finished using the {{GPUBuffer}} before the application can access its content.
+Once the {{GPUBuffer}} is mapped the application can synchronously ask for access
+to ranges of its content with {{GPUBuffer/getMappedRange}}. A mapped {{GPUBuffer}}
+cannot be used by the GPU and must be unmapped using {{GPUBuffer/unmap}} before
+it can be used on the [=Queue timeline=].
+
+Issue(gpuweb/gpuweb#605): Add client-side validation that a mapped buffer can
+  only be unmapped and destroyed on the worker on which it was mapped.
 
-Issue: Add client-side validation that a mapped buffer can only be unmapped and destroyed on the worker on which it was mapped.
+### <dfn method for=GPUBuffer>mapAsync(start, offset)</dfn> ### {#GPUBuffer-mapAsync}
 
-### {{GPUBuffer/mapReadAsync|GPUDevice.mapReadAsync}} ### {#GPUBuffer-mapReadAsync}
+<div algorithm="GPUBuffer.mapAsync">
 
-<div algorithm="GPUBuffer.mapReadAsync">
-    Issue: Handle error buffers once we have a description of the error monad.
+  **Arguments:**
+    - {{GPUSize64}} |offset|
+    - {{GPUSize64}} |size|
 
-    1. If the {{[[usage]]}} slot of |this| doesn't contain the {{GPUBufferUsage/MAP_READ}} bit or
-        if {{[[state]]}} isn't [=buffer state/unmapped=]:
+  **Returns:** {{Promise}}
 
-        1. Record a validation error on the current scope.
-        1. Return [=a promise rejected with=] an {{AbortError}}.
+  Issue(gpuweb/gpuweb#605): Handle error buffers once we have a description of the error monad.
 
-        Issue: Specify that the rejection happens on the device timeline.
+  1. If |size| is 0 and |offset| is less than |this|.{{[[size]]}}:
 
-    1. Let |p| be a new {{Promise}}.
-    1. Set the {{[[mapping]]}} slot of |this| to |p|.
-    1. Set the {{[[state]]}} slot of |this| to [=buffer state/mapping pending for reading=].
-    1. Enqueue an operation on the [=Queue timeline=] that will execute the following:
+    1. Set |size| to |this|.{{[[size]]}} - |offset|
 
-        1. Let |m| be a new {{ArrayBuffer}} of size the {{[[size]]}} of |this|.
-        1. Set the content of |m| to the content of |this|'s allocation.
-        1. Set the {{[[state]]}} slot of |this| to [=buffer state/mapped for reading=].
-        1. If |p| is pending:
+  1. If this call doesn't follow the [$mapAsync Valid Usage$]:
 
-            1. Resolve |p| with |m|.
+    1. Record a validation error on the current scope.
+    1. Return [=a promise rejected with=] an {{AbortError}} on the [=Device timeline=].
 
-    1. Return |p|.
+  1. Let |p| be a new {{Promise}}.
+  1. Set |this|.{{[[mapping]]}} to |p|.
+  1. Set |this|.{{[[state]]}} to [=buffer state/mapping pending=].
+  1. Enqueue an operation on the [=Queue timeline=] that will execute the following:
```

Comment:
The queue timeline belongs to a queue. Which queue in this case?

Reply:
There's a single queue at the moment so that's on the default queue. But when there is multiple queue, it will be either the queue that "owns" the resource, or waiting for all previous queue operations (on all queues) to finish if there's shared ownership (like readonly).

Point taken that this will need to be more clearly specified if we have multiqueue.

Reply:
I mean, today, we'd say that the default queue is implicitly used. This is implied and may be not obvious to the reader, but important.